lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -97,6 +97,29 @@ static LogicalResult lower(firrtl::ConstantOp op, ArrayRef<Value> operands,
   return success();
 }
 
+static LogicalResult lower(firrtl::WireOp op, ArrayRef<Value> operands,
+                           ConversionPatternRewriter &rewriter) {
+  auto resType = op.result().getType();
+  if (auto resFirType = resType.dyn_cast<FIRRTLType>()) {
+    auto resultType = RTLTypeConverter::convertType(resFirType);
+
+    if (auto intType = resultType.getValue().dyn_cast<IntegerType>()) {
+      rewriter.replaceOpWithNewOp<rtl::WireOp>(op, intType, op.nameAttr());
+    }
+    // TODO: Add support for vectors in RTL. The below code to handle FIRRTL
+    // vectors is not a solution.
+    else if (auto fvType = resFirType.dyn_cast<FVectorType>()) {
+      unsigned numElems = fvType.getNumElements();
+      auto elemType = RTLTypeConverter::convertType(fvType.getElementType());
+      unsigned width = elemType.getValue().dyn_cast<IntegerType>().getWidth();
```

Comment:
nit: dyn_cast may return null. If you are sure that the type is the one that you expect, then you should use cast<> instead. cast<T> internally asserts that the object isa<T>.

This also isn't checking that the Optional returned by `convertType` is non-null. (Note: you could use operator-> to access the value instead of getValue).

Reply:
Yep, River is right here.  Please fix both.

Incidentally, I'd split vector support out to a separate series of patches like you were planning to - just emit an error when `convertType` returns `None` and return failure.  This makes sure that the pass correctly bails out when it is given something that it doesn't handle.  Thanks!